frappe/desk/page/user_profile/user_profile_controller.js

```diff
@@ -15,23 +15,16 @@ class UserProfile {
 	}
 
 	show() {
-		let route = frappe.get_route();
-		this.user_id = route[1] || frappe.session.user;
```

Comment:
If this line is removed it takes away ability to view other's profile 😄 

Making this dynamic action would be simplest/elegant solution IMO: https://github.com/frappe/frappe/blob/3baa09401e1d3abd8ff14b88b7b0a8b53257b87a/frappe/utils/install.py#L184-L190 

Alternative: when user id isn't there just assume session user id but don't update the route.

Reply:
ah, we can i think - had the same doubt but it seems like we just set the `user_id` to be that particular user when we view a different profile and render from there itself

https://github.com/frappe/frappe/blob/develop/frappe/desk/page/user_profile/user_profile_controller.js#L63-L82

https://github.com/frappe/frappe/blob/b3ea89670cb9f9b65d417b73b37e427384e64f0d/frappe/desk/page/user_profile/user_profile_controller.js#L37-L38

Reply:
I meant links to other users like `/app/user-profile/john` will stop working since it's not using route parameter anymore.

Reply:
ah, yes. can i know why do we need that?

Reply:
"need" is a strong word 😆 It used to work before, it will stop working now, that's all. 

I think the simplest fix is to not update the route and set user_id to session id when route doesn't contain userid. That won't break anything as far as I can tell.

Reply:
for making it easier to view other people's profile?

Reply:
ah cool, sorry i missed your comment, didn't refresh the page 😅